pinot-core/src/main/java/org/apache/pinot/core/io/writer/impl/MutableOffHeapByteArrayStore.java

```diff
@@ -113,41 +106,41 @@ private int add(byte[] value) {
         // full
         return -1;
       }
-      for (int i = 0, j = startOffset; i < value.length; i++, j++) {
-        _byteBuffer.put(j, value[i]);
-      }
-      _byteBuffer.putInt(_numValues * Integer.BYTES, startOffset);
+      _pinotDataBuffer.readFrom(startOffset, value);
```

Comment:
I take it that this is the batch read being referred to? Probably I am missing something, but it seemed that the underlying implementations were still iterating over individual bytes. If so, where is the speedup coming from?

Reply:
The underlying implementation is simply iterating over individual bytes. Check PinotByteBuffer for details.

Reply:
I'm also curious if this gives performance improvement and how much. As per the code in `org.apache.pinot.core.segment.memory.PinotByteBuffer#readFrom(long, byte[], int, int)`, if the buffer size is >10 (hard-coded value), the buffer is indeed duplicated during the batch operation so there could be garbage impact too.